### PR TITLE
Next: Login & Registration form error/success handling and validation

### LINKED ISSUE #5377: Login & Registration form error/success handling and validation
#### Errors
Use VSF Error Handling to display errors in login/registration forms.

How the flow should look like:

1. User is entering the data to the form
1. Sends the form
1. UI notification with "Something went wrong!" is being displayed
1. If the error corresponds to a certain field display it under this fields (eg email already taken under email field) (so you should do ifs based on error message therefore login/registration form are parts of integration templates)
1. For other issues add `SfAlert` on top of registration/login form with a friendly message. For example

```html
<SfAlert v-if="error.logIn.message === 'Lorem ipsum dolor'" type="danger" :message="error.logIn.message" />
```

#### Success
* If user logs in display "Successfully logged in" notification of type `success`
* If user registers successfully display "Successfully created a new account"

## What are the acceptance criteria 

* [ ] All errors are handled for Logging in according to specification
* [ ] All errors are handled for Registering according to specification
* [ ] Success messages are displayed

### PR DESCRIPTION
**Pull request by lukaszjedrasik**
### Related Issues

closes #5377 

### Short Description of the PR

- Added VSF Error handling to display errors in Login/Registration forms.
- Added `AuthModal` component and splitted`LoginModal` component to `LoginForm` and `RegisterForm`.
- Changed `isLoginModal` and `toggleLoginModal` to `isAuthModal` and `toggleAuthModal`

### Screenshots of Visual Changes before/after (if There Are Any)

**Desktop:**
https://user-images.githubusercontent.com/40273258/113286914-1c9eb680-92ed-11eb-9e05-5e0e87736225.mp4

**Mobile:**
https://user-images.githubusercontent.com/40273258/113286961-32ac7700-92ed-11eb-803c-aec2ad6a986c.mp4

### Pull Request Checklist

- [x] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [x] I have tested my Pull Request on production build and (to my knowledge) it works without any issues

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

File: packages/commercetools/theme/components/Auth/RegisterForm.vue [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        type="danger"
+        :message="$t(registerError.message)"
+      />
+      <form class="form" @submit.prevent="handleSubmit(handleRegister)" autocomplete="off">
+        <ValidationProvider rules="required|email" v-slot="{ errors }">
+          <SfInput
+            data-cy="login-input_email"
+            v-model="form.email"
+            :valid="registerError && registerError.field && registerError.field === 'email' ? false : !errors[0]"
+            :errorMessage="registerError && registerError.field && registerError.field === 'email' ? $t(registerError.message) : $t(errors[0])"
```
Comment (andrzejewsky) [inline packages/commercetools/theme/components/Auth/RegisterForm.vue:15]: create computed properties for both

File: packages/commercetools/theme/components/Auth/RegisterForm.vue [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      if (!registerError.value) {
+        send({
+          type: 'success',
+          message: $i18n.t('Successfully created a new account')
+        });
+        toggleAuthModal();
+        return;
+      }
+
+      if (registerError.value && registerError.value.code === 'DuplicateField' && registerError.value.message.search(form.value.email) !== -1) {
```
Comment (andrzejewsky) [inline packages/commercetools/theme/components/Auth/RegisterForm.vue:135]: jeezzz, why that's sooo long? :D is it readable? can you create a function for checking this or even a computed property?

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

**Comment by andrzejewsky** [comment]
solve conflicts please

**Comment by bloodf** [comment]
This is closed due to inactivity.

If you want your PR to be re-opened, please update your files, check for any conflicts.

If your PR is for Vue Storefront 1 target the new repository at https://github.com/vuestorefront/vue-storefront-1

**Pull request closed without merging**